packages/synthetix-main/contracts/storage/Distribution.sol

```diff
@@ -89,151 +137,137 @@ library Distribution {
         dist.totalShares = dist.totalShares + sharesUint128 - actor.shares;
 
         actor.shares = sharesUint128;
-        actor.lastValuePerShare = shares == 0 ? int128(0) : dist.valuePerShare;
+
+        actor.lastValuePerShare = newActorShares == 0 ? int128(0) : dist.valuePerShare;
     }
 
     /**
-     * @dev Updates an actor's value in the distribution to the specified amount.
+     * @dev Updates an actor's individual value in the distribution to the specified amount.
      *
-     * The incoming value is translated into a number of shares,
-     * which modify the total number of shares in the distribution, and thus
-     * the distribution's totalValue.
+     * The change in value is manifested in the distribution by changing the actor's number of shares in it, and thus the distribution's total number of shares.
      *
-     * Returns the amount by which the distribution's number of shares changed.
-     *
-     * Note: A distribution is intended to either use updateActorShares or updateActorValue, but not both
-     * during the distribution's lifetime.
+     * Returns the resulting amount of shares that the actor has after this change in value.
      */
-    function updateActorValue(
+    function setActorValue(
         Data storage dist,
         bytes32 actorId,
-        int value
-    ) internal returns (uint shares) {
+        int newActorValue
+    ) internal returns (uint resultingShares) {
+        // This functions requires that the distribution has a valuePerShare and totalShares to calculate the actor's number of shares with the incoming value.
         if (dist.valuePerShare == 0 && dist.totalShares != 0) {
             revert ZeroValuePerShare();
         }
 
         DistributionActor.Data storage actor = dist.actorInfo[actorId];
 
+        // TODO DBQuestion: why do we need that the actor was previously initialized (actor.share =/= 0)
         if (actor.lastValuePerShare != 0) {
             revert InconsistentDistribution();
         }
 
-        // Calculate the number of shares that the
-        // change in value produces.
+        // Represent the actor's change in newActorValue by changing the actor's number of shares,
+        // and keeping the distribution's valuePerShare constant.
 
-        // If the distribution is empty, set valuePerShare to 1.0,
-        // and the number of shares to the given value.
-        if (dist.totalShares == 0) {
-            dist.valuePerShare = DecimalMath.UNIT_PRECISE_INT128;
-            shares = (value > 0 ? value : -value).int256toUint256(); // Ensure value is positive
-        }
-        // If the distribution is not empty, the number of shares
-        // is determined by the valuePerShare.
-        else {
-            // Calculate number of shares this way instead of value / valuePerShare,
-            // in order to avoid precision loss.
-            shares = uint((value * int128(dist.totalShares)) / totalValue(dist));
-        }
```

Comment:
Again, this is also semantic overloading. Moving it to a separate `_getSharesForValue()` function, which is used internally this time.